wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1812,12 +1814,14 @@ impl super::Adapter {
             0, 0, 0, 0,
         ];
 
+        let drop_guard = drop_callback.map(|callback| crate::DropGuard { callback });
```

Comment:
Might be worth a crate::DropGuard::from_option which takes an Option<DropCallback> and returns Option<DropGuard> to make this code a bit simpler